modules/vcf2maf/1.3/vcf2maf.smk

```diff
@@ -137,66 +161,162 @@ rule _vcf2maf_gnomad_filter_maf:
         touch {output.dropped_maf}
         """)
 
-def get_chain(wildcards):
-    if "38" in str({wildcards.genome_build}):
-        return reference_files("genomes/{genome_build}/chains/grch38/hg38ToHg19.over.chain")
+def get_original_genome(wildcards):
+    # Determine the original (i.e. input) reference genome for this sample
+    # Since this module projects to various output genome builds, we need to parse the sample table for the starting build
+    # To determine what we need to do
+    runs_table = config['lcr-modules']["vcf2maf"]["runs"]
+    sample_entry = runs_table.loc[(runs_table["tumour_sample_id"] == wildcards.tumour_id) & (runs_table["normal_sample_id"] == wildcards.normal_id) & (runs_table["tumour_seq_type"] == wildcards.seq_type)]
+    if len(sample_entry) == 0:
+        raise AttributeError("Unable to locate a a sample with tumour_id:{wildcards.tumour_id}, normal_id:{wildcards.normal_id}, seq_type:{wildcards.seq_type} in the \'runs\' table")
+    original_genome_build = sample_entry.iloc[0]["tumour_genome_build"]
+    return original_genome_build
+
+def get_chain(genome_build):
+    # NOTE: This only currently supports hg38 and hg19. If you are using other genome builds, this will need to be handled
+    if VCF2MAF_GENOME_VERSION[genome_build] == "grch38":
+        return reference_files("genomes/" + genome_build + "/chains/grch38/hg38ToHg19.over.chain")
```

Comment:
does this work if no reference files are present and need to be generated _de novo_?

Reply:
I haven't explicitly tested this, but it should work. At the end of the day the subworkflow is just getting a string for a desired file; this shouldn't matter if this results from wildcard expansion or string concatenation